stackpanel/src/main/java/org/vaadin/addons/stackpanel/client/StackPanelConnector.java

```diff
@@ -44,14 +47,16 @@ protected void extend(ServerConnector target) {
         DOM.insertChild(panel.captionNode, captionToggle, 2);
 
         updateStyleName(panel);
+       
 
         panel.addDomHandler(new ClickHandler() {
 
             @Override
             public void onClick(ClickEvent event) {
                 EventTarget eventTarget = event.getNativeEvent().getEventTarget();
-                if (eventTarget.cast() == panel.captionNode
-                        || eventTarget.cast() == panel.captionNode.getFirstChildElement()) {
+           
+                if (eventTarget.cast() == panel.captionNode || isChildofNode(panel, eventTarget)
+                		|| eventTarget.cast() == captionToggle.getChild(0))  { 
```

Comment:
Adding this condition should be enough to fix the bug (or `eventTarget.cast() == captionToggle.getFirstChildElement()`).
Unless there is some use case that I'm unaware, 
I think it's better to target the specific DOM element for the toggling functionality. The loop through `panel.captionNode` will prevent future extension (e.i. If someone wants to make an Action Button Panel extension, it will also trigger the toggle functionality).